### PR TITLE
feat: adds datadog logging

### PR DESCRIPTION
**Pull request by kevinperaza**
## Description

- feat: adds datadog logging

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change
-

### CODE REVIEW

File: src/utils/logging.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  getSystemName,
+  getSystemVersion,
+  getVersion,
+  isEmulator,
+} from 'react-native-device-info';
+
+type AttributeMap = Record<string, string>;
+
+export const logger = (() => {
+  const ddTok = 'pubb96b84a13912504f4354f2d794ea4fab';
```
Comment (jleon15) [inline src/utils/logging.ts:14]: i know this is a public token but is it fine hardcoding it like this?

Comment (kevinperaza) [reply src/utils/logging.ts:14]: That's a very good question, I just ported what we did in [ios](https://github.com/Basis-Theory/basistheory-ios/blob/cef523f4ea59f1d208bc968a40fcc3c6581136c1/BasisTheoryElements/Sources/BasisTheoryElements/TelemetryLogging.swift#L14) but we should add them to our secrets

Comment (kevinperaza) [reply src/utils/logging.ts:14]: That's a very good question, I just ported what we did in [ios](https://github.com/Basis-Theory/basistheory-ios/blob/cef523f4ea59f1d208bc968a40fcc3c6581136c1/BasisTheoryElements/Sources/BasisTheoryElements/TelemetryLogging.swift#L14) but we should add them to our secrets at some point(?)

**Review by jleon15 — APPROVED** [review]

**Pull request merged**